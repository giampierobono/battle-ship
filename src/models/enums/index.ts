export * from './board-mode.enum';
export * from './boat-type.enum';
export * from './position-state-enum';
