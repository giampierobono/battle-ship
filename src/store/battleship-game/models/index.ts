export * from './set-game-player-turn-payload.model';
export * from './save-player-move-payload.model';
export * from './set-winner-player-payload.model';
