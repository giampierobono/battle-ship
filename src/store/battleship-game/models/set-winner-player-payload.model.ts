export interface SetWinnerPlayerPayloadModel {
  winnerPlayerIndex: number;
}
