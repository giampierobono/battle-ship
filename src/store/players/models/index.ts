export * from './add-players-payload.model';
export * from './create-player-payload.model';
