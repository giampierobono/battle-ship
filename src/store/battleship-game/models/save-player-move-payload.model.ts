import { PositionStateEnum, ShotPositionModel } from '../../../models';

export interface SavePlayerMovePayloadModel {
  shotPosition: ShotPositionModel;
  state: PositionStateEnum;
}
