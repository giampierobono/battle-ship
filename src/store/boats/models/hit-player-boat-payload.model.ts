import { ShotPositionModel } from '../../../models';

export interface HitPlayerBoatPayloadModel {
  boatIndex: number;
  position: ShotPositionModel;
}
