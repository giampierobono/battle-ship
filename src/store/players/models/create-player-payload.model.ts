import { PlayerModel } from '../../../models';

export interface CreatePlayerPayloadModel {
  newPlayer: PlayerModel;
}
