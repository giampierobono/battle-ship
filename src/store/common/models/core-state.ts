import { IPlayersState } from '../../players';
import { IBoatsState } from '../../boats';
import { IBoatShapeState } from '../../boats-shapes';
import { IGameSettingsState } from '../../game-settings';
import { IBattleShipGameState } from '../../battleship-game';

export interface CoreState {
  players: IPlayersState;
  boats: IBoatsState;
  boatShapes: IBoatShapeState;
  settings: IGameSettingsState;
  game: IBattleShipGameState;
}
