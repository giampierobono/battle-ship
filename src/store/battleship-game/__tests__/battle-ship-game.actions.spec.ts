import {
  addSunkBoat,
  nextTurn,
  savePlayerMove,
  setBoardPositionStatus,
  setPlayerTurn,
  setWinnerPlayer,
  tryShoot,
} from '../battle-ship-game.actions';
import { PositionStateEnum } from '../../../models';

describe('Battleship game actions', () => {
  describe('setPlayerTurn', () => {
    it('should return correct action structure', () => {
      expect(setPlayerTurn({ playerIndex: 1 })).toMatchSnapshot();
    });
  });

  describe('tryShoot', () => {
    it('should return correct action structure', () => {
      expect(tryShoot({ playerIndex: 0, row: 0, column: 0 })).toMatchSnapshot();
    });
  });

  describe('setBoardPositionStatus', () => {
    it('should return correct action structure', () => {
      expect(setBoardPositionStatus({ playerIndex: 0, row: 0, column: 0 })).toMatchSnapshot();
    });
  });

  describe('savePlayerMove', () => {
    it('should return correct action structure', () => {
      expect(
        savePlayerMove({ state: PositionStateEnum.HIT, shotPosition: { playerIndex: 1, row: 0, column: 0 } })
      ).toMatchSnapshot();
    });
  });

  describe('nextTurn', () => {
    it('should return correct action structure', () => {
      expect(nextTurn()).toMatchSnapshot();
    });
  });

  describe('addSunkBoat', () => {
    it('should return correct action structure', () => {
      expect(addSunkBoat({ boatIndex: 0, sunkByPlayerIndex: 0, boatBelongsToPlayerIndex: 1 })).toMatchSnapshot();
    });
  });

  describe('setWinnerPlayer', () => {
    it('should return correct action structure', () => {
      expect(setWinnerPlayer({ winnerPlayerIndex: 1 })).toMatchSnapshot();
    });
  });
});
