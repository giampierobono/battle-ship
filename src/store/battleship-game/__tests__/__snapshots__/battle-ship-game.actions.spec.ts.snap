// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Battleship game actions addSunkBoat should return correct action structure 1`] = `
Object {
  "boatBelongsToPlayerIndex": 1,
  "boatIndex": 0,
  "sunkByPlayerIndex": 0,
  "type": "[Battleship game] New sunk boat",
}
`;

exports[`Battleship game actions nextTurn should return correct action structure 1`] = `
Object {
  "type": "[Battleship game] Move the next player",
}
`;

exports[`Battleship game actions savePlayerMove should return correct action structure 1`] = `
Object {
  "shotPosition": Object {
    "column": 0,
    "playerIndex": 1,
    "row": 0,
  },
  "state": "HIT",
  "type": "[Battleship game] Save player move",
}
`;

exports[`Battleship game actions setBoardPositionStatus should return correct action structure 1`] = `
Object {
  "column": 0,
  "playerIndex": 0,
  "row": 0,
  "type": "[Battleship game] Set board position status",
}
`;

exports[`Battleship game actions setPlayerTurn should return correct action structure 1`] = `
Object {
  "playerIndex": 1,
  "type": "[Battleship game] Set player turn",
}
`;

exports[`Battleship game actions setWinnerPlayer should return correct action structure 1`] = `
Object {
  "type": "[Battleship game] Set winning player",
  "winnerPlayerIndex": 1,
}
`;

exports[`Battleship game actions tryShoot should return correct action structure 1`] = `
Object {
  "column": 0,
  "playerIndex": 0,
  "row": 0,
  "type": "[Battleship game] Player fires a shot",
}
`;
